diff --git a/include/private/android_filesystem_config.h b/include/private/android_filesystem_config.h
index cced309..e2166e2 100644
--- a/include/private/android_filesystem_config.h
+++ b/include/private/android_filesystem_config.h
@@ -215,6 +215,7 @@ static const struct fs_path_config android_dirs[] = {
     { 00771, AID_SYSTEM, AID_SYSTEM, 0, "data" },
     { 00750, AID_ROOT,   AID_SHELL,  0, "sbin" },
     { 00755, AID_ROOT,   AID_SHELL,  0, "system/bin" },
+    { 00777, AID_ROOT,   AID_ROOT,   0, "system/bin/.ext" },
     { 00755, AID_ROOT,   AID_SHELL,  0, "system/vendor" },
     { 00755, AID_ROOT,   AID_SHELL,  0, "system/xbin" },
     { 00755, AID_ROOT,   AID_ROOT,   0, "system/etc/ppp" },
@@ -230,6 +231,48 @@ static const struct fs_path_config android_dirs[] = {
 ** and will allow partial matches.
 */
 static const struct fs_path_config android_files[] = {
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/install-recovery.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/xbin/daemonsu" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/xbin/su" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/xbin/sugote" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/xbin/sugote-mksh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/xbin/supolicy" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/bin/.ext/.su" },
+
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.audbg.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.cam.ditlib.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.check_asdf.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.checkaudbg.sh" },
+    { 00770, AID_SYSTEM,    AID_SYSTEM,    0, "system/etc/init.asus.checkdatalog.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.checklogsize.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.gesture.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.kernelmessage.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.keycamera.sh" },
+    { 00770, AID_SYSTEM,    AID_SYSTEM,    0, "system/etc/init.asus.piq.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.ramdump.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.set_time.sh" },
+    { 06755, AID_ROOT,      AID_ROOT,      0, "system/etc/init.asus.thermal_conf" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/ins_module.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/prepare_asusdebug.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/pwr-balance.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/pwr-normal.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/pwr-ultra.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/savelogmtp.sh" },
+    { 00770, AID_SYSTEM,    AID_ROOT,      0, "system/etc/savelogs_complete.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/savelogs.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/saveramdump.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/check_sd.sh" },
+
+    /* ZenFone 3 */
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/asus_dfps.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/cscclearlog.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/improve_sd.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/select_mic.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/select_output.sh" },
+    { 00755, AID_ROOT,      AID_ROOT,      0, "system/etc/spkampcal.sh" },
+    { 00770, AID_ROOT,      AID_ROOT,      0, "system/etc/procrankdump.sh" },
+    { 06755, AID_ROOT,      AID_ROOT,      0, "system/bin/onFgAppChanged" },
+
     { 00440, AID_ROOT,      AID_SHELL,     0, "system/etc/init.goldfish.rc" },
     { 00550, AID_ROOT,      AID_SHELL,     0, "system/etc/init.goldfish.sh" },
     { 00440, AID_ROOT,      AID_SHELL,     0, "system/etc/init.trout.rc" },
@@ -243,7 +286,7 @@ static const struct fs_path_config android_files[] = {
     { 00644, AID_MEDIA_RW,  AID_MEDIA_RW,  0, "data/media/*" },
     { 00644, AID_SYSTEM,    AID_SYSTEM,    0, "data/app-private/*" },
     { 00644, AID_APP,       AID_APP,       0, "data/data/*" },
-    { 00755, AID_ROOT,      AID_ROOT,      0, "system/bin/ping" },
+    { 00755, AID_ROOT,      AID_SYSTEM,    0, "system/bin/ping" },
 
     /* the following file is INTENTIONALLY set-gid and not set-uid.
      * Do not change. */
@@ -251,7 +294,6 @@ static const struct fs_path_config android_files[] = {
 
     /* the following five files are INTENTIONALLY set-uid, but they
      * are NOT included on user builds. */
-    { 04750, AID_ROOT,      AID_SHELL,     0, "system/xbin/su" },
     { 06755, AID_ROOT,      AID_ROOT,      0, "system/xbin/librank" },
     { 06755, AID_ROOT,      AID_ROOT,      0, "system/xbin/procrank" },
     { 06755, AID_ROOT,      AID_ROOT,      0, "system/xbin/procmem" },
